lib/weakhash/weakhash.go

```diff
@@ -45,7 +49,7 @@ func Find(ir io.Reader, hashesToFind []uint32, size int) (map[uint32][]int64, er
 
 	offsets := make(map[uint32][]int64)
 	for _, hashToFind := range hashesToFind {
-		offsets[hashToFind] = nil
+		offsets[hashToFind] = make([]int64, 0, maxWeakhashFinderHits)
```

Comment:
Mmm. Back of envelope calculation says this allocates ~640 KiB per GiB of input file (1024*1024/128*80), which should be fine.